File: design-patterns/src/main/java/org/prateek/SOLIDPrinciples/ISP/BadCode/MultiPurposeMachine.java
package org.Prathamesh.SOLIDPrinciples.ISP.BadCode;

public class MultiPurposeMachine implements Machine{
    @Override
    public void print(Document doc) {
        System.out.println("Printing document...");

    }

    @Override
    public void scan(Document doc) {
        System.out.println("Scan document...");

    }

    @Override
    public void copy(Document doc) {
        System.out.println("Copy document...");
    }
}

File: cpp/SOLIDPrinciples/ISP/BadCode/MultiPurposeMachine.cpp
#ifndef MULTIPURPOSE_MACHINE_CPP
#define MULTIPURPOSE_MACHINE_CPP

#include <iostream>
#include "Machine.cpp"

class MultiPurposeMachine : public Machine {
public:
    void print(Document* doc) override {
        std::cout << "Printing document..." << std::endl;
    }

    void scan(Document* doc) override {
        std::cout << "Scan document..." << std::endl;
    }

    void copy(Document* doc) override {
        std::cout << "Copy document..." << std::endl;
    }
};

#endif 